### PR TITLE
Added support for project organization policy

### PR DESCRIPTION
**Pull request by rdayama-eng**
```release-note:none

```

### CODE REVIEW

**Comment by rdayama-eng** [comment]
@rileykarson Please take a look

File: third_party/validator/cai.go [hunk 1]
```diff
@@ -41,6 +42,36 @@ type IAMBinding struct {
 	Members []string `json:"members"`
 }
 
+type OrgPolicy struct {
+	Constraint     string                 `json:"constraint,omitempty"`
+	ListPolicy     *Policy_ListPolicy     `json:"listPolicy"`
+	BooleanPolicy  *Policy_BooleanPolicy  `json:"booleanPolicy"`
+	RestoreDefault *Policy_RestoreDefault `json:"restoreDefault"`
+	UpdateTime     *Timestamp             `json:"update_time,omitempty"`
```
[the remark this answers is not in the document]
Comment (rdayama-eng) [reply third_party/validator/cai.go:50]: Yes this is according to the CAI model

File: third_party/validator/cai.go [hunk 2]
```diff
@@ -41,6 +42,36 @@ type IAMBinding struct {
...
+	Seconds int64 `json:"seconds,omitempty"`
+	Nanos   int64 `json:"nanos,omitempty"`
+}
+
+type Policy_ListPolicy_AllValues int32
+
+type Policy_ListPolicy struct {
+	AllowedValues     []string                    `json:"allowed_values,omitempty"`
+	DeniedValues      []string                    `json:"denied_values,omitempty"`
+	AllValues         Policy_ListPolicy_AllValues `json:"all_values,omitempty"`
```
[the remark this answers is not in the document]
Comment (rdayama-eng) [reply third_party/validator/cai.go:63]: This is according to the CAI model

File: third_party/validator/cai.go [hunk 3]
```diff
@@ -41,6 +42,36 @@ type IAMBinding struct {
...
+	RestoreDefault *Policy_RestoreDefault `json:"restoreDefault"`
+	UpdateTime     *Timestamp             `json:"update_time,omitempty"`
+}
+
+type Timestamp struct {
+	Seconds int64 `json:"seconds,omitempty"`
+	Nanos   int64 `json:"nanos,omitempty"`
+}
+
+type Policy_ListPolicy_AllValues int32
```
[the remark this answers is not in the document]
Comment (rdayama-eng) [reply third_party/validator/cai.go:58]: This is according to the CAI model

File: third_party/validator/cai.go [hunk 4]
```diff
@@ -14,6 +14,7 @@ type Asset struct {
 	Type      string         `json:"asset_type"`
 	Resource  *AssetResource `json:"resource,omitempty"`
 	IAMPolicy *IAMPolicy     `json:"iam_policy,omitempty"`
+	OrgPolicy []*OrgPolicy   `json:"org_policy,omitempty"`
```
[the remark this answers is not in the document]
Comment (rdayama-eng) [reply third_party/validator/cai.go:17]: Org policies are tied to project, folder and organization. According to CAI model orgpolicy is an attribute of Asset.
This PR is only for supporting project organization policy. Terraform Resource - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_organization_policy#version.

File: third_party/validator/cai.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rileykarson) [reply third_party/validator/cai.go:58]: Is naming the Go values specific things important? That's not clear to me.

File: third_party/validator/cai.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rileykarson) [reply third_party/validator/cai.go:17]: Got it. Thanks!

**Review by rileykarson — COMMENTED** [review]
I'll trust your judgement on the `OrgPolicy` subtypes.

Have you verified that this works with Terraform Validator? If not, are we able to do that before merging here? also /cc @morgante

File: third_party/validator/project_organization_policy.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
+package google
```
Comment (rileykarson) [inline third_party/validator/project_organization_policy.go:1]: Make sure to add this file to https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform_object_library.rb#L76

Comment (rdayama-eng) [reply third_party/validator/project_organization_policy.go:1]: Added path to org policy file

**Comment by rileykarson** [comment]
@rdayama-eng, I'm interested if you have a response to my prior question:

> Have you verified that this works with Terraform Validator? If not, are we able to do that before merging here?

Also /gcbrun

**Comment by rdayama-eng** [comment]
> I'll trust your judgement on the `OrgPolicy` subtypes.
> 
> Have you verified that this works with Terraform Validator? If not, are we able to do that before merging here? also /cc @morgante

I have verified it works with terraform validator. Verified using integ test https://github.com/GoogleCloudPlatform/terraform-validator/blob/b4fd0d644053d51f9d8565f6d59b4cc8c858cf67/test/read_test.go#L31.  @morgante Please have a look

**Comment by morgante** [comment]
@rdayama-eng Just to confirm, did you add a new integration test for org policies? It might be good to open a draft PR there first for review.

**Comment by rdayama-eng** [comment]
> @rdayama-eng Just to confirm, did you add a new integration test for org policies? It might be good to open a draft PR there first for review.

Sure, I will do it and link it here

**Review by rileykarson — APPROVED** [review]
Code LGTM- mind linking the test before I hit merge?

**Comment by rdayama-eng** [comment]
Terraform Validator PR https://github.com/GoogleCloudPlatform/terraform-validator/pull/157

**Pull request merged**